rest/handler.go

```diff
@@ -400,38 +407,61 @@ func (h *handler) checkAuth(context *db.DatabaseContext) (err error) {
 	return nil
 }
 
-func (h *handler) checkAdminAuth(dbContext *db.DatabaseContext, checkPermissions []string, storePermissionsResults bool) error {
+func checkAdminAuth(serverContext *ServerContext, dbContext *db.DatabaseContext, basicAuthUsername, basicAuthPassword string, checkPermissions []string) (map[string]bool, error) {
```

Comment:
Seems cleaner to pass in a slice of management endpoints/httpClient given only one of serverContext or dbContext will be used to obtain them.

Reply:
Wanted to encapsulate the admin api work in the same function rather than have that management endpoint work separated.